examples_tests/41.VisibilityBuffer/occlusionCullMap.comp

```diff
@@ -0,0 +1,34 @@
+#version 460 core
+
+#extension GL_EXT_shader_16bit_storage : require
+
+#include "cullShaderCommon.h"
+layout(local_size_x = WORKGROUP_SIZE) in;
+
+#define ENABLE_OCCLUSION_CULLED_COMMAND_BUFFER
+#define ENABLE_VISIBLE_BUFFER
+#define ENABLE_CUBE_DRAW_GUID_BUFFER
+#define ENABLE_CUBE_COMMAND_BUFFER
+#define OCCLUSION_CULLED_COMMAND_BUFF_SET 0
+#define OCCLUSION_CULLED_COMMAND_BUFF_BINDING 0
+#define VISIBLE_BUFF_SET 0
+#define VISIBLE_BUFF_BINDING 1
+#define CUBE_DRAW_GUID_BUFF_SET 0
+#define CUBE_DRAW_GUID_BUFF_BINDING 2
+#define CUBE_COMMAND_BUFF_SET 0
+#define CUBE_COMMAND_BUFF_BINDING 3
+#include "occlusionCullingShaderCommon.glsl"
+
+void main()
+{
+    if(gl_GlobalInvocationID.x > cubeIndirectDraw.draw.instanceCount)
+        return;
+
+    if(uint(visibleBuff.visible[gl_GlobalInvocationID.x]) == 0)
+        return;
+    
+    const uint batchMdiOffset = cubeDrawGUIDBuffer.drawGUID[gl_GlobalInvocationID.x];
```

Comment:
why do you need this map?, you could have passed `batchMdiOffset` as an extra per-instance attribute (just like the MVP) instead of `instanceID` to occlusion cull fragment shader

Reply:
nvm, your approach makes more sense (no gaps in indirect dispatch)